client/components/TestPlanUpdater/TestPlanUpdaterModal.jsx

```diff
@@ -0,0 +1,541 @@
+                    testResult
+                );
+
+                const saveMutation = testResult.completedAt
+                    ? SUBMIT_TEST_RESULT_MUTATION
+                    : SAVE_TEST_RESULT_MUTATION;
+
+                const { data: savedData } = await client.mutate({
+                    mutation: saveMutation,
+                    variables: {
+                        testResultId: copiedTestResultInput.id,
+                        testResultInput: copiedTestResultInput
+                    }
+                });
+
+                if (savedData.errors) throw savedData.errors;
+
+                // Set the number of created results
+                setLoadingSpinnerProgress(prevState => ({
+                    ...prevState,
+                    numerator: prevState.numerator + 1
+                }));
+            }
+        }
+
+        setSafeToDeleteReportId(currentReportId);
+
+        if (deleteChecked) {
+            setShowDeleteTestPlan(true);
+            setAlertCompletion({
+                message:
+                    'New report created without errors. Are you sure you want to delete the old report?',
+                success: false,
+                visible: true
+            });
+        } else {
+            setShowDeleteTestPlan(false);
+            setAlertCompletion({
+                message: 'Completed without errors.',
+                success: true,
+                visible: true
+            });
+        }
+
+        setLoadingSpinnerProgress(prevState => ({
+            ...prevState,
+            visible: false
+        }));
+    };
+
+    const closeAndDeleteOldTestPlan = async () => {
+        if (newReport) {
+            if (deleteChecked) {
+                await deleteOldTestPlanReport();
+            }
+            await triggerTestPlanReportUpdate();
+        }
+
+        handleClose();
+    };
+
+    const deleteOldTestPlanReport = async () => {
+        setAlertCompletion({
+            visible: true,
+            success: false,
+            message: 'Deleting old report...'
+        });
+        await client.mutate({
+            mutation: DELETE_TEST_PLAN_REPORT,
+            variables: {
+                testPlanReportId: safeToDeleteReportId
+            }
+        });
+        setSafeToDeleteReportId(null);
+        setAlertCompletion({
+            visible: true,
+            success: true,
+            message: 'Old report deleted.'
+        });
+    };
+
+    const deleteComponent = (
+        <Form.Check type="checkbox" className="delete-checkbox">
+            <Form.Check.Input
+                type="checkbox"
+                onChange={() => {
+                    setDeleteChecked(!deleteChecked);
+                }}
+                checked={deleteChecked}
+            />
+            <Form.Check.Label className="delete-label">
+                <FontAwesomeIcon icon={faTrashAlt} className="trash-icon" />{' '}
+                Delete old Test Plan
+            </Form.Check.Label>
```

Comment:
Clicking the text didn't check the checkbox. I guess you might need to either put the checkbox inside the label or add an `id` prop to the checkbox and a `for` prop for the label.